### PR TITLE
Added a blending in delay for antags and a small refactor

### PR DESCRIPTION
**Pull request by farie82**
**What does this PR do:**
>Necaladun
Adding a default objective to most antags that consists of "Blending in" for a delayed amount (5-10 minutes). When the delay ends, they get their real objectives.

Does exactly that. Clings, vampires and agents now get a 5 till 10 minute delay before getting objectives.
It will notify them that they are said antag and that they will receive objectives within 10 minutes.
Only matters for roundstart antags.

**Changelog:**
:cl:
balance: Changelings, vampires and agents now get a 5 till 10 minute delay before getting their objectives.
/:cl:

### CODE REVIEW

File: code/game/gamemodes/vampire/vampire.dm [hunk 1]
```diff
@@ -180,19 +184,14 @@
 	dat += {"To bite someone, target the head and use harm intent with an empty hand. Drink blood to gain new powers.
 You are weak to holy things and starlight. Don't go into space and avoid the Chaplain, the chapel and especially Holy Water."}
 	to_chat(vampire.current, dat)
-	to_chat(vampire.current, "<B>You must complete the following tasks:</B>")
+	to_chat(vampire.current, "<B>The Syndicate will contact you within 10 minutes to give you your objectives. Lay low till then.</B>")
```
Comment (TDSSS) [inline code/game/gamemodes/vampire/vampire.dm]: I don't think vampires get their objectives from the syndicate, unless they are the rare vamp tators.

Comment (TDSSS) [reply code/game/gamemodes/vampire/vampire.dm]: I have just learned that it is actually in the lore that vamps and clings work for the Syndicate. Huh.

**Comment by ScratKnapp** [comment]
..why

**Comment by farie82** [comment]
Mostly to prevent rushes. More a problem at low pop than on high pop.

**Comment by SkeletalElite** [comment]
My main problem with this is that you can't really even start preparing in that time, especially as a traitor. Vampires and changelings, whatever they know they need blood or genomes anyways, but a traitor has no idea what theyre going to have to do, and the steps you take to prepare if you are about to hijack are way different than if you have to steal the advanced boots or something like that.

**Comment by Kyep** [comment]
Does this inform people they're antags right away, but delay their objectives for a few minutes?
Or does it not even set them as antags for a few minutes?

**Comment by farie82** [comment]
@kyep it'll inform them that they are an antag and that they will get objectives within 10 minutes. Ill update the description

**Comment by Kyep** [comment]
I'm opposed to this PR.

The main justification I've heard for this PR is that it reduces antag fishing. But, objectively, it doesn't, since antags still know they're antags as soon as they spawn in. Even if this PR changed antags to only be picked 5-10 minutes into a round, I still don't think that would be a significant disincentive either. Really, I think only job bans from antag roles are enough of a disincentive to actually make a dent in that problem.

The second justification I've heard for this PR is that it reduces the chances that antags will rush to complete their objectives in the first minutes of a round. I am skeptical that this is really a problem. I don't see it happening. Even when I have seen it happen, which is rare, it typically results in them being quickly caught, as their objectives are in someone's starting area, and those people are still there. So, they tend to be instantly seen and quickly caught. Lastly, even if rushes were a problem (I don't think they are) and even if starting positions don't discourage them enough (they do) I am skeptical that this is a good solution to the problem. It just means people will rush 5-10 minutes into rather than right away. In fact, I worry that having their objectives update 5-10 minutes in, with a notice, might prompt MORE antags to rush their objectives early than currently do it, because they're now being prompted and the prompt may encourage them to do so earlier than they otherwise would.
I also worry that technical considerations here could easily result in bugs. Or simply render the system largely irrelevant. Cultists, for example, know they're gonna need to cult people, so even hiding cult objectives won't eliminate that incentive to get started on it early. Traitors with documents know that one of their objectives will be to keep their documents, so they can start taking actions in that vein immediately. Changelings/vamps will know they need DNA/blood so they can again start doing those things immediately. Etc. Due to the nature of all these antags I just don't think this will have a meaningful positive impact.

For all of these reasons, I'm against this change.

**Comment by farie82** [comment]
> I'm opposed to this PR.
> 
> The main justification I've heard for this PR is that it reduces antag fishing. But, objectively, it doesn't, since antags still know they're antags as soon as they spawn in. Even if this PR changed antags to only be picked 5-10 minutes into a round, I still don't think that would be a significant disincentive either. Really, I think only job bans from antag roles are enough of a disincentive to actually make a dent in that problem.
> 
> The second justification I've heard for this PR is that it reduces the chances that antags will rush to complete their objectives in the first minutes of a round. I am skeptical that this is really a problem. I don't see it happening. Even when I have seen it happen, which is rare, it typically results in them being quickly caught, as their objectives are in someone's starting area, and those people are still there. So, they tend to be instantly seen and quickly caught. Lastly, even if rushes were a problem (I don't think they are) and even if starting positions don't discourage them enough (they do) I am skeptical that this is a good solution to the problem. It just means people will rush 5-10 minutes into rather than right away. In fact, I worry that having their objectives update 5-10 minutes in, with a notice, might prompt MORE antags to rush their objectives early than currently do it, because they're now being prompted and the prompt may encourage them to do so earlier than they otherwise would.
> 
> I also worry that technical considerations here could easily result in bugs. Or simply render the system largely irrelevant. Cultists, for example, know they're gonna need to cult people, so even hiding cult objectives won't eliminate that incentive to get started on it early. Traitors with documents know that one of their objectives will be to keep their documents, so they can start taking actions in that vein immediately. Changelings/vamps will know they need DNA/blood so they can again start doing those things immediately. Etc. Due to the nature of all these antags I just don't think this will have a meaningful positive impact.
> 
> For all of these reasons, I'm against this change.

The documents now get spawned in once they get their objectives. With a warning about it going to happen.
Other than that I mostly made this PR due to it being on the wanted changes list.

**Comment by Kyep** [comment]
I don't blame you. I actually appreciate the fact you were tackling items on wanted lists.
Normally I'm fine with items on other peoples' wanted lists.
This is a rare exception, an idea that just doesn't sit right with me at all.

**Comment by Fox-McCloud** [comment]
Not a fan of this either; sometimes you need to know, right away, what your'e doing, when you get antag.

Delaying when you get antag won't help much and will be very easily gameable for some individuals anyway "Hey, I was supposed to be an officer---oh. Antag. Will wait 5-10 before I get my objectives'

I'd like to know the underlying goal of making the "blend in" requirement I suppose.

**Comment by Kyep** [comment]
Sorry, but given that a majority of the active maints and headmins (3 of the 5) object to this PR, we'll be closing it.

**Comment by farie82** [comment]
Aight. Ill close it myself here.

**Pull request closed without merging**